custom/icds_reports/ucr/data_sources/tech_issue_cases.json

```diff
@@ -353,31 +284,50 @@
         "type": "property_name",
         "property_name": "aww_or_ls"
       },
-      "supervisor_id": {
+      "aww_or_ls_id": {
+        "type": "conditional",
         "test": {
           "operator": "eq",
-          "type": "boolean_expression",
           "expression": {
-            "datatype": null,
             "type": "property_name",
             "property_name": "aww_or_ls"
           },
+          "type": "boolean_expression",
           "property_value": "aww"
         },
         "expression_if_true": {
-          "location_id_expression": {
-            "datatype": null,
             "type": "property_name",
             "property_name": "aww_id"
-          },
-          "type": "location_parent_id"
         },
-        "type": "conditional",
         "expression_if_false": {
-          "datatype": "string",
-          "type": "property_name",
-          "property_name": "ls_id"
+            "type": "property_name",
+            "property_name": "ls_id"
         }
+      },
+      "supervisor_id": {
+          "location_id": {
+            "type": "conditional",
+            "test": {
+              "operator": "eq",
+              "expression": {
+                "type": "property_name",
+                "property_name": "aww_or_ls"
+              },
+              "type": "boolean_expression",
+              "property_value": "aww"
+            },
+            "expression_if_true": {
+                "type": "property_name",
+                "property_name": "aww_id"
+            },
+            "expression_if_false": {
+                "type": "property_name",
+                "property_name": "ls_id"
+            }
+          },
+          "location_type": "supervisor",
+          "location_propery": "_id",
+          "type": "ancestor_location"
```

Comment:
I don't think this will work here. In the case of an LS the location won't be part of the ancestry and the expression doesn't consider the initial location. I think the best thing to do would be to move the `ancestor_location` expression to the `expression_if_true` clause.

Reply:
That's a good catch, fixed it.